pkg/client/unversioned/clientcmd/merged_client_builder.go

```diff
@@ -115,10 +115,10 @@ func (config *DeferredLoadingClientConfig) ClientConfig() (*restclient.Config, e
 		// TODO: this shouldn't be a global - the client config rules should be
 		//   handling this.
 		defaultConfig, defErr := DefaultClientConfig.ClientConfig()
-		if IsConfigurationInvalid(defErr) && !IsEmptyConfig(err) {
+		if defErr != nil && !IsConfigurationInvalid(defErr) {
 			return mergedConfig, nil
```

Comment:
First, define what you mean by "non-host client" so I can make sure I'm following your scenario.  What inputs have been provided in your container?

Default config is rolled into the merged config (hence why it is called merged) before it is returned to us.  We only fetch the default config to verify that default config (which is assumed to not be the result of explicit user input) matches the returned mergedConfig (which means it does not appear the user has provided a flag or config file that varies from the default).  That's the existing behavior from 1.3.0, except for the security gap we uncovered that meant that *only* kubectl should have a default config.  Everyone else should get an empty default config.

The change I made for that was to the merge behavior for kubectl - if you can explain why your default config *isnt* invalid (everyone except kubectl should have empty default config) that may help identify the problem.